test/zmumu_pp5tev/tp_from_aod_Data.py

```diff
@@ -0,0 +1,629 @@
+    src = cms.InputTag("patMuonsWithTrigger"),
+    cut = cms.string("pt > 15 && abs(eta)<2.4 && "+MuonIDFlags.TightId.value()+" && ("
+       +    "!triggerObjectMatchesByPath('HLT_HIL2Mu3_NHitQ10_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL3Mu3_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL3Mu3_NHitQ10_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL2Mu5_NHitQ10_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL3Mu5_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL3Mu5_NHitQ10_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL2Mu7_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL3Mu7_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL2Mu12_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL3Mu12_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL2Mu15_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL3Mu15_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIMu15_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIMu17_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL2Mu20_v*',1,0).empty()"
+       + "|| !triggerObjectMatchesByPath('HLT_HIL3Mu20_v*',1,0).empty())"
+       + " && pfIsolationR04().sumChargedHadronPt/pt < 0.2"),
+)
+process.pseudoTag = cms.EDFilter("MuonSelector",
+    src = cms.InputTag("muons"),
+    cut = cms.string("pt > 15 && isGlobalMuon && numberOfMatchedStations >= 2 && pfIsolationR04().sumChargedHadronPt/pt < 0.2")
+)
+if TRIGGER == "DoubleMu":
+    process.tagMuons.cut = ("pt > 6 && (isGlobalMuon || isTrackerMuon) && isPFMuon "+
+                            " && !triggerObjectMatchesByCollection('hltL3MuonCandidates').empty()"+
+                            " && pfIsolationR04().sumChargedHadronPt/pt < 0.2")
+
+process.oneTag  = cms.EDFilter("CandViewCountFilter", src = cms.InputTag("tagMuons"), minNumber = cms.uint32(1))
+
+process.probeMuons = cms.EDFilter("PATMuonSelector",
+    src = cms.InputTag("patMuonsWithTrigger"),
+    cut = cms.string("track.isNonnull"),  # no real cut now
+)
+
+process.tpPairs = cms.EDProducer("CandViewShallowCloneCombiner",
+    #cut = cms.string('60 < mass < 140 && abs(daughter(0).vz - daughter(1).vz) < 4'),
+    cut = cms.string('60 < mass && abs(daughter(0).vz - daughter(1).vz) < 4'),
+    decay = cms.string('tagMuons@+ probeMuons@-')
+)
+process.onePair = cms.EDFilter("CandViewCountFilter", src = cms.InputTag("tpPairs"), minNumber = cms.uint32(1))
+
+from MuonAnalysis.TagAndProbe.muon.tag_probe_muon_extraIso_cff import ExtraIsolationVariables
+
+from MuonAnalysis.TagAndProbe.puppiIso_cfi import load_fullPFpuppiIsolation
+process.fullPuppIsolationSequence = load_fullPFpuppiIsolation(process)
+from MuonAnalysis.TagAndProbe.puppiIso_cff import PuppiIsolationVariables
+
+process.tpTree = cms.EDAnalyzer("TagProbeFitTreeProducer",
```

Comment:
If the tree becomes too large, we could consider removing some of the variables and flags, since it looks like there is a lot of info that might not be needed

Reply:
true, puppi info not needed. will remove